langext/intro.txt

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[intro]]
+== Introduction
+
+This document describes language extensions that can be supported optionally by the compiler in addition to the functionality from the existing standard language versions. To detect whether extension is supported by the compiler special feature macros can be used `\__opencl_c_<feature name>`. Each feature macro has a value corresponding to the publishing date. Some extensions are available in all versions and others only in the specific versions of OpenCL kernel language (see <<lang_ext_list,the full list of lanuage extension>>).
+
+
+[[lang_ext_list]]
+== Language Extension List
+ 
+The section contains all optional language extensions listed along with their feature macros as well as OpenCL version in which the extensions are available .
+
+// Editors note: Please keep this table in alphabetical order!
+
+[cols="5,4,2,4",options="header",]
+|====
+| *Extension Name*
+| *Feature macro*
+| *FM Value*
+| *OpenCL versions*
+
+| <<opencl_c_variadic_macro,Variadic Macros>>
+| __opencl_c_variadic_macro
```

Comment:
Should the feature macro be named `__opencl_variadic_macro`, which leaves the door open for them being used in C++ for OpenCL as well?  Or is there a reason to prefer `__opencl_c_variadic_macro` explicitly?

Reply:
OK, that works.  Have you given any thought what would happen if a language extension did apply to C++ for OpenCL as well?